LabExT/Movement/Stage3DSmarAct.py

```
@@ -0,0 +1,610 @@
+        ]
+
+    def move_relative(self, x, y):
+        """Performs a relative movement by x and y. Specified in units of micrometers.
+
+        Parameters
+        ----------
+        x : int
+            Movement in x direction by x measured in um.
+        y : int
+            Movement in y direction by y measured in um.
+        """
+        self.logger.debug(
+            'Want to relative move %s to x = %s um and y = %s um',
+            self.address,
+            x,
+            y)
+
+        self.channels[Axis.X].move(diff=x, mode=MovementType.RELATIVE)
+        self.channels[Axis.Y].move(diff=y, mode=MovementType.RELATIVE)
+
+    def move_absolute(self, pos):
+        """Performs an absolute movement to the specified position in units of micrometers.
+
+        Parameters
+        ----------
+        position : list
+            Position in [x,y] format measured in um
+        """
+        self.logger.debug(
+            'Want to absolute move %s to x = %s um and y = %s um',
+            self.address,
+            pos[0],
+            pos[1])
+
+        self.channels[Axis.X].move(
+            diff=pos[0], mode=MovementType.ABSOLUTE)
+        self.channels[Axis.Y].move(
+            diff=pos[1], mode=MovementType.ABSOLUTE)
+
+    # Helper methods
+
+    def _open_system(self):
+        handle = ct.c_ulong()
+        if self._exit_if_error(
+            MCSC.SA_OpenSystem(
+                handle,
+                self.address,
+                bytes('sync', 'utf-8'))):
+            return handle
+        return None
+
+    def _raise_if_sensor_non_linear(self) -> None:
+        for index, channel in self.channels.items():
+            if not channel.is_sensor_linear:
+                raise RuntimeError(
+                    'Channel {} of stage {} has no supported linear sensor!'.format(
+                        index.name, self.address))
+        self.logger.debug("Linear x, y and z sensor present")
+
+    def _exit_if_error(self, status: int) -> bool:
```

Discussion point: I kept this method similar to [PiezaStage.py](https://github.com/LabExT/LabExT/blob/main/LabExT/Movement/PiezoStage.py#L151). Only a boolean return value I introduced, but we don't really "exit" the program here. Should we raise an exception here so that the program stops?

---

Yes, I would suggest to immediately raise a RuntimeError with a telling error message.

---

I agree. Will change it 👍🏼